src/domain/hooks/helpers/event-cache.helper.ts

```diff
@@ -0,0 +1,553 @@
+        promises.push(
+          ...this.onTransactionEventDeletedMultisigTransaction(event),
+        );
+        break;
+      case TransactionEventType.MODULE_TRANSACTION:
+        promises.push(...this.onTransactionEventModuleTransaction(event));
+        break;
+
+      case TransactionEventType.EXECUTED_MULTISIG_TRANSACTION:
+        promises.push(
+          ...this.onTransactionEventExecutedMultisigTransaction(event),
+        );
+        break;
+      case TransactionEventType.NEW_CONFIRMATION:
+        promises.push(...this.onTransactionEventNewConfirmation(event));
+        break;
+      case TransactionEventType.INCOMING_ETHER:
+        promises.push(...this.onTransactionEventIncomingEther(event));
+        break;
+      case TransactionEventType.OUTGOING_ETHER:
+        promises.push(...this.onTransactionEventOutgoingEther(event));
+        break;
+      case TransactionEventType.INCOMING_TOKEN:
+        promises.push(...this.onTransactionEventIncomingToken(event));
+        break;
+      case TransactionEventType.OUTGOING_TOKEN:
+        promises.push(...this.onTransactionEventOutgoingToken(event));
+        break;
+      case TransactionEventType.MESSAGE_CREATED:
+        promises.push(...this.onTransactionEventMessageCreated(event));
+        break;
+      case TransactionEventType.MESSAGE_CONFIRMATION:
+        promises.push(...this.onTransactionEventMessageConfirmation(event));
+        break;
+      case ConfigEventType.CHAIN_UPDATE:
+        promises.push(...this.onConfigEventChainUpdate(event));
+        break;
+      case ConfigEventType.SAFE_APPS_UPDATE:
+        promises.push(...this.onConfigEventSafeAppsUpdate(event));
+        break;
+      case TransactionEventType.SAFE_CREATED:
+        promises.push(...this.onTransactionEventSafeCreated(event));
+        break;
+    }
+    return Promise.all(promises);
+  }
+
+  public onEventLog(event: Event): void {
+    switch (event.type) {
+      case TransactionEventType.PENDING_MULTISIG_TRANSACTION:
+      case TransactionEventType.DELETED_MULTISIG_TRANSACTION:
+      case TransactionEventType.EXECUTED_MULTISIG_TRANSACTION:
+      case TransactionEventType.NEW_CONFIRMATION:
+        this._logSafeTxEvent(event);
+        break;
+      case TransactionEventType.MODULE_TRANSACTION:
+      case TransactionEventType.INCOMING_ETHER:
+      case TransactionEventType.OUTGOING_ETHER:
+      case TransactionEventType.INCOMING_TOKEN:
+      case TransactionEventType.OUTGOING_TOKEN:
+        this._logTxEvent(event);
+        break;
+      case TransactionEventType.MESSAGE_CREATED:
+      case TransactionEventType.MESSAGE_CONFIRMATION:
+        this._logMessageEvent(event);
+        break;
+      case ConfigEventType.CHAIN_UPDATE:
+      case ConfigEventType.SAFE_APPS_UPDATE:
+        this._logEvent(event);
+        break;
+      case TransactionEventType.SAFE_CREATED:
+        break;
+    }
+  }
+
+  // Transaction Service events
```

Comment:
TODO: Split the service into multiple classes, each handling a specific type of event, to adhere to the Single Responsibility Principle.

Reply:
Whilst I understand the approach, experimenting with a secondary class resulted in a lot of boilerplate and it's the [`Event` entity](https://github.com/safe-global/safe-client-gateway/blob/refactor-hooks-repository/src/routes/hooks/entities/schemas/event.schema.ts#L17) is being handled (although it consists of both Config Service and Transaction Service `type`s).

What are your thoughts? Do you think it is essential we separate the class? If so, I'll add a TODO comment for a followup to keep the diff. minimal.